mmv1/products/alloydb/api.yaml

```diff
@@ -0,0 +1,286 @@
+# Copyright 2021 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Product
+name: Alloydb
+display_name: AlloyDB
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://alloydb.googleapis.com/v1beta/
+scopes:
+  - https://www.googleapis.com/auth/cloud-identity
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: AlloyDB API
+    url: https://console.cloud.google.com/apis/library/alloydb.googleapis.com
+objects:
+  - !ruby/object:Api::Resource
+    name: "Cluster"
+    self_link: "projects/{{project}}/locations/{{location}}/clusters/{{cluster_id}}"
+    base_url: "projects/{{project}}/locations/{{location}}/clusters"
+    create_url: "projects/{{project}}/locations/{{location}}/clusters?clusterId={{cluster_id}}"
+    update_verb: :PATCH
+    update_mask: true
+    min_version: beta
+    description: "A managed alloydb cluster."
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        "AlloyDB": "https://cloud.google.com/alloydb/docs/"
+      api: "https://cloud.google.com/alloydb/docs/reference/rest/v1beta/projects.locations.clusters/create"
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        path: "name"
+        base_url: "{{op_id}}"
+        wait_ms: 1000
+        timeouts: !ruby/object:Api::Timeouts
+          insert_minutes: 10
+          update_minutes: 10
+          delete_minutes: 10
+      result: !ruby/object:Api::OpAsync::Result
+        path: "response"
+      status: !ruby/object:Api::OpAsync::Status
+        path: "done"
+        complete: true
+        allowed:
+          - true
+          - false
+      error: !ruby/object:Api::OpAsync::Error
+        path: "error"
+        message: "message"
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: "clusterId"
+        required: true
+        input: true
+        url_param_only: true
+        description: |
+          The ID of the alloydb cluster.
```

Comment:
Could we switch to using name as id pattern. I know there is president for this current design but I think conceptually it's not as intuitive for the end user.

ref: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/binaryauthorization/api.yaml#L26

Reply:
Taking over review - I disagree & would prefer clusterId, to match the API more closely.